certora/ArrangerConduit.spec

```diff
@@ -0,0 +1,860 @@
+
+    bool canCall = roles.canCall(ilk, e.msg.sender, currentContract, to_bytes4(0xd2543ccb)); // requestFunds(bytes32,address,uint256,string)
+    mathint requestedFunds = requestedFunds(asset, ilk);
+    mathint totalRequestedFunds = totalRequestedFunds(asset);
+    mathint numRequests = getFundRequestsLength();
+
+    requestFunds@withrevert(e, ilk, asset, amount, info);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = !canCall;
+    bool revert3 = requestedFunds + amount > max_uint256;
+    bool revert4 = totalRequestedFunds + amount > max_uint256;
+
+    assert lastReverted <=> revert1 || revert2 || revert3 ||
+                            revert4, "Revert rules failed";
+} 
+
+// Verify correct storage changes for non reverting requestFunds
+rule cancelFundRequest(uint256 fundRequestId) {
+    env e;
+
+    // general check before
+    mathint numRequestsBefore = getFundRequestsLength();
+
+    // actual (asset,ilk) for actual request Id before
+    IArrangerConduit.FundRequest requestBefore = getFundRequest(fundRequestId);
+    bytes32 infoHashBefore = aux.hashString(requestBefore.info);
+
+    mathint requestedFundsBefore = requestedFunds(requestBefore.asset, requestBefore.ilk);
+    mathint totalRequestedFundsBefore = totalRequestedFunds(requestBefore.asset);
+
+    // other (asset,ilk) for actual request Id before
+    bytes32 otherIlk;
+    address otherAsset;
+    require otherIlk != requestBefore.ilk || otherAsset != requestBefore.asset;
+    mathint requestedFundsOtherBefore = requestedFunds(otherAsset, otherIlk);
+
+    address otherAsset2;
+    require otherAsset2 != requestBefore.asset;
+    mathint totalRequestedFundsOtherBefore = totalRequestedFunds(otherAsset2); 
+
+    // other request Id before
+    uint256 otherIndex;
+    require otherIndex != fundRequestId;
+
+    IArrangerConduit.FundRequest requestOtherBefore = getFundRequest(otherIndex);
+    bytes32 infoHashOtherBefore = aux.hashString(requestOtherBefore.info);
+
+    cancelFundRequest(e, fundRequestId);
+
+    // general check after
+    mathint numRequestsAfter = getFundRequestsLength();
+
+    // actual (asset,ilk) for actual request Id after
+    IArrangerConduit.FundRequest requestAfter = getFundRequest(fundRequestId);
+    bytes32 infoHashAfter = aux.hashString(requestAfter.info);
+
+    mathint requestedFundsAfter = requestedFunds(requestBefore.asset, requestBefore.ilk);
+    mathint totalRequestedFundsAfter= totalRequestedFunds(requestBefore.asset);
+
+    // other (asset,ilk) for actual request Id before
+    mathint requestedFundsOtherAfter = requestedFunds(otherAsset, otherIlk);
+    mathint totalRequestedFundsOtherAfter = totalRequestedFunds(otherAsset2); 
+
+    // other request Id after
+    IArrangerConduit.FundRequest requestOtherAfter = getFundRequest(otherIndex);
+    bytes32 infoHashOtherAfter = aux.hashString(requestOtherBefore.info);
+
+    // general asserts
+    assert numRequestsAfter == numRequestsBefore, "num requests changed";
```

Comment:
Probably this one should be evaluated in storageChanges rule and just in a specific rule when it actually changes.

Reply:
@oldchili let me know what you want to do with this one, if you consider is not worth it will resolve it.